internal/cache.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 Tigris Data, Inc.
```

Comment:
Is this generic cache backend or for real-time?

Reply:
This looks like something specific to real-time so probably the file shouldn't be cache.go

Reply:
I agree, I think it would make sense to have a generic cache layer and then the realtime stream on top of that.

Reply:
There is a cache backend that is inside `store/cache` and the real-time stream will be on top of it. This file is the serialized data that is stored inside Cache for two use cases. 

The first one is for the Cache backend where the usage is - Get/Set/etc, the other is if someone wants to leverage Cache streams. The StreamData currently has some leaks because it has some states that the channel needs which needs some refactoring which will be done as part of the next diff which is built on top of this with real-time stuff.